SO-4446: Update to Java 17 and Eclipse IDE 2021-12

apeteri opened the pull request.

apeteri committed:
SO-4446: Update JRE to JavaSE-17 in project build configurations

apeteri committed:
SO-4446: Update execution environment to JavaSE-17 in bundle manifests

apeteri committed:
SO-4446: Update Tycho to 2.4.0-SNAPSHOT

- Set JavaSE-16 as the execution environment for
target-platform-configuration
- Set Java source and target version to 16 for tycho-compiler-plugin and
maven-compiler-plugin
SO-4446: Use Tycho 2.6.0-SNAPSHOT, set JDK release to 17 for compilers

apeteri committed:
SO-4446: Update mockito to 3.11.2 from 2.13.0

apeteri committed:
SO-4446: Update typetools to 0.6.3 from 0.4.8

apeteri committed:
SO-4446: Update jacoco to 0.8.7 from 0.8.6

apeteri committed:
SO-4446: Update JRE to JavaSE-17 in launch configurations

- Remove references to CMS collector
- Remove unsupported AlwaysLockClassLoader option
- Add --add-opens options for base modules java.lang.reflect, java.lang,
java.util, java.time

apeteri committed:
SO-4446: Add com.diffplug.osgi.extension.sun.misc to dependencies

apeteri committed:
SO-4446: Update JustJ JRE feature version to 17.0.1 release

apeteri committed:
SO-4446: Update startup scripts

- Use built-in JRE if JAVA_HOME is not defined
- Remove deprecated AlwaysLockClassLoader option
- Remove CMS garbage collector arguments
- Add --add-opens arguments for modules java.lang.reflect, java.lang,
java.util, java.time

apeteri committed:
SO-4446: Use JDK 17 Temurin in maven GitHub workflow

apeteri committed:
SO-4446: Work around "Illegal type in constant pool" error when...

...running Groovy validation scripts

apeteri committed:
SO-4446: Add explicit import statement for Record to avoid conflicts...

...with java.lang.Record, added in JDK 16

apeteri committed:
SO-4446: Update setup-java action to 2.2.0

apeteri committed:
feat(setup): Update project setup model

- Use Eclipse 2021-12 for committers as the base product (not specified
in model)
- Increase heap to 2048m
- Update M2e to 1.18.2 (for enhanced target platform capabilities)
- Update Groovy-Eclipse to 4.4.0
- Set import statement threshold to 5 for static and non-static imports
- Remove setup tasks related to local target platform generation

apeteri committed:
feat(releng): Add Maven dependencies directly to target definition file

Support for bundles in Maven repositories renders use of
pde-target-maven-plugin, the dependency list in the root POM and the
"consider pomDependencies" setting obsolete.

apeteri committed:
feat(releng): Remove unused properties; bump Groovy version in root POM

apeteri committed:
refactor(commons): Rename project to avoid M2e test project detection

In newer releases of M2e, when synchronizing Maven projects with their
POM that are also Eclipse plug-ins, source folders are converted to test
sources if the project name includes "tests" anywhere in the string, or
ends with "test" or "tests".

According to this rule, c.b.s.test.commons will not have test sources,
but the source folder of c.b.index.tests.tools will be converted to
test-only on synchronization; since the former bundle depends on the
latter, a build path error of "Project has only main sources but depends
on project ... which has only test sources" appears.

Changing the project name allows us to keep the source folder a regular
one without having to fight the tooling constantly.

apeteri committed:
chore(classpath): Commit updated build path settings for projects...

...that depend on artifacts from Maven

apeteri committed:
chore(api-docs): Include "name" property in package-lock.json

apeteri committed:
chore(validation.snomed): Allow M2e to remove Groovy DSL support from...

...the build path, and set the source folder to regular from test-only

apeteri committed:
chore(core): Update Groovy to 2.5.15 in target platform

apeteri committed:
chore: Update Groovy to 2.5.15 in bundle manifest files

apeteri committed:
refactor(restassured): Remove deprecated REST-assured wrapper bundle

apeteri committed:
feat(hamcrest): Add compatibility bundle for Hamcrest version 2

apeteri committed:
fix(test.commons): Add default object mapper to REST-assured config

The classpath scanning-based mechanism does not find the Jackson2
backend after upgrading to 4.0, however if a default mapper is
registered, it can use it without any issue.

apeteri committed:
chore(launch): Update launch configuration of e2e tests

apeteri committed:
chore(classpath): Move remaining JREs to 17, sort classpath entries

apeteri committed:
chore(manifests): Move remaining BREEs to 17; fix revert of bundle...

...properties following an incorrect merge conflict resolution

apeteri committed:
chore(pom): Restore parent module version following an incorrect...

...merge conflict resolution

apeteri committed:
feat(commons): Move commons-collections4 version to a variable

apeteri committed:
chore(bin): Update launch script files

- Update JustJ JRE version in Java executable paths
- Synchronize Equinox launcher version in CIS scripts
- Remove CMS options from CIS scripts
- Add --add-opens options to CIS scripts

apeteri committed:
docs(pom): Update comment above allowed target platform version ranges

apeteri committed:
fix(setup): Revise update site and versions for the M2E tooling

Eclipse 2021-12 already ships with version 1.19.0. Also updated the
corresponding section in README.md.

apeteri committed:
fix(hamcrest): Remove version element from pom.xml so that target...

...platform computation succeeds at the beginning of the build

apeteri committed:
feat(setup): Use Groovy 3.0.9 for Java 17 compatibility

- Setup model
- README.md instructions
- Target platform bundles
- Version filter in root pom.xml

apeteri committed:
chore(target): Remove commons-lang 2, update version of commons-lang3

apeteri committed:
chore(target): Update REST-assured from 4.0.0 to 4.3.0

apeteri committed:
chore(target): Add objenesis, a dependency of hamcrest 2

apeteri committed:
chore(deps): Add bytebuddy as a required bundle to mockito

apeteri committed:
chore(deps): Depend on Groovy 3.0.9 in MANIFEST.MF files

apeteri committed:
fix(test.commons): Add workaround for Jackson 2 detection; move...

...ObjectMapper configuring class to a separate file

apeteri committed:
fix(validation.snomed): Remove extra comma from rule266.groovy

apeteri committed:
chore(deps): Update launch configurations

apeteri committed:
chore(deps): Keep B2i repository in root POM

This seems to be used for resolving the mvn: URL in the target
definition file for ECL at build time; in the IDE these can only be
resolved against repositories in settings.xml.

apeteri committed:
Merge branch '8.x' into feature/SO-4446_java_17_eclipse_2021_12

apeteri committed:
chore(deps): Update SLF4J and Logback dependencies in target platform

apeteri committed:
fix(readme): Update instruction related to JDK versions

apeteri committed:
chore(build): Update Tycho to 2.6.0 release

apeteri committed:
Merge remote-tracking branch 'origin/8.x' into feature/SO-4446_java_17_eclipse_2021_12

cmark reviewed (approved):
LGTM!

apeteri committed:
Merge remote-tracking branch 'origin/8.x' into
feature/SO-4446_java_17_eclipse_2021_12

Conflicts:
	pom.xml

cmark commented:
This requires additional changes to the base docker image we offer for the example docker file. Keep this PR open until all infra and dependant repository issues are resolved.

apeteri committed:
build(target-platform): Remove obsolete .gitignore file

Target platform definition files are no longer generated locally, so all
file changes should be considered to be committed in project
"target-platform".

apeteri committed:
chore(snow-owl): Sort version properties in root POM alphabetically

apeteri committed:
chore(snow-owl): Move Tycho's Director, Publisher and Version...

...plug-in configuration to the pluginManagement section; they don't
need to run for all artifact types and will be selected automatically or
manually in a child module POM.

apeteri committed:
chore(snow-owl): Move jgit.ignore value to a single line

There is (or was) some problem with including whitespace in this
section, see https://bugs.eclipse.org/bugs/show_bug.cgi?id=436325 .

apeteri committed:
chore(snow-owl): Add comment to indicate why Maven repository entries...

...are still required

apeteri committed:
chore(target-platform): Remove Hamcrest 1.3.0 from target definition

apeteri committed:
feat(setup): Update Xtend and Xtext to 2.25.0, MWE to 2.12.1

apeteri committed:
feat(product): Remove property "serverMode" from product extension

It is unused, the corresponding code from ApplicationContext has been
deleted earlier.

apeteri committed:
feat(snow-owl): Remove property for Xtend compiler version from root POM

There are no references to it in any of the child modules.

apeteri committed:
feat(target-platform): Remove Groovy-Eclipse from the target platform

Groovy is fetched from Maven central instead, so none of the features
provided by Groovy-Eclipse are required in the server build and TP.

apeteri committed:
feat(core.dependencies.feature): Replace included Groovy bundles

apeteri committed:
fix(target-platform): Use HTTPS URL for EMF repository

apeteri committed:
fix(server.update): Sort auto-start bundles list alphabetically

cmark reviewed (approved):
LGTM!

apeteri committed:
Synchronize Eclipse plug-in list with Setup model in README.md

nagyo committed:
ci(docker): use base docker image without pre-installed JDK
